### PR TITLE
Create notification service

### LINKED ISSUE #109: Create email notification service

### PR DESCRIPTION
**Pull request by bogdangavrylyak**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by BorysMalynovskyi — COMMENTED** [review]
Unrelated files included

File: CarBackEnd/Car.BLL/Services/Implementation/GoogleDriveService.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public Task<string> DeleteFile(string fileId)
         /// <returns>All files</returns>
         public async Task<IEnumerable<File>> GetAllFiles()
         {
-           return (await service.Files.List().ExecuteAsync()).Files;
+            return (await service.Files.List().ExecuteAsync()).Files;
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Services/Implementation/GoogleDriveService.cs:48]: please revert extra space

File: CarBackEnd/Car.BLL/Services/Implementation/SmtpClientService.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        }
+
+        public async Task SendAsync(MimeMessage mailMessage, EmailConfiguration emailConfiguration)
+        {
+            using (var client = new SmtpClient())
+            {
+                try
+                {
+                    await client.ConnectAsync(emailConfiguration.SmtpServer, emailConfiguration.Port, true);
+                    client.AuthenticationMechanisms.Remove("XOAUTH2");
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Services/Implementation/SmtpClientService.cs:26]: the string value is hardcoded, move to const

File: CarBackEnd/Car.BLL/Services/Implementation/EmailSenderService.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            _smtpClient = smtpClient;
+            _razorViewToStringRenderer = razorViewToStringRenderer;
+        }
+
+        public async Task CancelJourneyAsync(Message message)
+        {
+            for (int i = 0; i < message.Recipients.Count; i++)
+            {
+                string body = await _razorViewToStringRenderer.RenderViewToStringAsync(
+               "/Views/Emails/CancelJourney/CancelJourney.cshtml",
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Services/Implementation/EmailSenderService.cs:34]: move this path to const

File: CarBackEnd/Car.BLL/Services/Implementation/EmailSenderService.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+            message.Recipients.ForEach(recipient => mailMessage.To.Add(recipient.Address));
+
+            AlternateView alternateView = AlternateView
+                .CreateAlternateViewFromString(message.Content, null, "text/html");
+
+            if (message.Attachments != null)
+            {
+                LinkedResource lr = new LinkedResource(message.Attachments.OpenReadStream());
+                lr.ContentId = "imgpath";
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Services/Implementation/EmailSenderService.cs:65]: hardcode

File: CarBackEnd/Car.BLL/Services/Implementation/EmailSenderService.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+               "/Views/Emails/CancelJourney/CancelJourney.cshtml",
+               new CancelJourneyViewModel
+               {
+                   DriverName = message.DriverAddress.Name,
+                   TimeOfStoppage = message.CancelDate,
+                   Attachments = message.Attachments,
+               });
+
+                message.Content = body;
+                message.Subject = $"Your journey has been cancelled";
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Services/Implementation/EmailSenderService.cs:43]: suggest move messages to the const class, ex EmailMessageConstants

File: CarBackEnd/Car.BLL/Services/Implementation/EmailSenderService.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            };
+
+            message.Recipients.ForEach(recipient => mailMessage.To.Add(recipient.Address));
+
+            AlternateView alternateView = AlternateView
+                .CreateAlternateViewFromString(message.Content, null, "text/html");
+
+            if (message.Attachments != null)
+            {
+                LinkedResource lr = new LinkedResource(message.Attachments.OpenReadStream());
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Services/Implementation/EmailSenderService.cs:64]: please use the full names of variables: linkedResource

File: CarBackEnd/CarBackEnd/CarBackEnd.csproj [hunk 7]
```diff
@@ -5,9 +5,13 @@
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
-    <DocumentationFile>C:\Users\Nazar\Cource\SoftServe\SoftServeCar\Car-Back-End\CarBackEnd\CarBackEnd\CarBackEnd.xml</DocumentationFile>
+    <DocumentationFile>F:\SoftServeCarProject\BackEnd7\Car-Back-End\CarBackEnd\CarBackEnd\CarBackEnd.xml</DocumentationFile>
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/CarBackEnd/CarBackEnd.csproj:8]: this config is hardcoded

File: CarBackEnd/CarBackEnd/CarBackEnd.csproj [hunk 8]
```diff
@@ -5,9 +5,13 @@
...
   </PropertyGroup>
 
+  <ItemGroup>
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/CarBackEnd/CarBackEnd.csproj:11]: do we need to add this folder?

File: CarBackEnd/CarBackEnd/Controllers/UserController.cs [hunk 9]
```diff
@@ -42,7 +42,7 @@ public IActionResult GetUserById(int userId)
         [HttpPut("{userId}/avatar")]
         public async Task<IActionResult> UploadUserAvatar(int userId, [FromForm] FormImage userFile)
         {
-           return Ok(await imageService.UploadImage(userId, userFile.image));
+            return Ok(await imageService.UploadImage(userId, userFile.image));
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/CarBackEnd/Controllers/UserController.cs:45]: please revert spaces

File: CarBackEnd/Car.BLL/Services/Interfaces/IEmailSenderService.cs [hunk 10]
```diff
@@ -0,0 +1,10 @@
+﻿using Car.BLL.Dto.Email;
+using System.Threading.Tasks;
+
+namespace Car.BLL.Services.Interfaces
+{
+    public interface IEmailSenderService
+    {
+        Task CancelJourneyAsync(Message mailingMessage);
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Services/Interfaces/IEmailSenderService.cs:8]: please add a unit test coverage

File: CarBackEnd/Car.BLL/Services/Interfaces/ISmptClient.cs [hunk 11]
```diff
@@ -0,0 +1,11 @@
+﻿using Car.BLL.Dto.Email;
+using MimeKit;
+using System.Threading.Tasks;
+
+namespace Car.BLL.Services.Interfaces
+{
+    public interface ISmptClient
+    {
+        Task SendAsync(MimeMessage mailMessage, EmailConfiguration emailConfiguration);
```
Comment (OleksandrAntoniuk) [inline CarBackEnd/Car.BLL/Services/Interfaces/ISmptClient.cs:9]: please add a unit test coverage

**Review by OleksandrAntoniuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**